File: leet_easy_c++/leet_088/leet_088a.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        if(n==0) return; 
        if(m==0){
            nums1 = nums2;
            return;
        }
        for(int i=m-1, j=n-1, k=m+n-1; k>=0; k--){
            if(nums2[j]>=nums1[i]){
                nums1[k]=nums2[j];
                j--;
            }
            else{
                nums1[k]=nums1[i];
                i--;
            }
            if(j<0) return;
            if(i<0){
                k--;
                for(; j>=0;j--,k--)
                    nums1[k]=nums2[j];
            return;
            }
        }    
    }
};

// correct and simple implementation!

File: leet_easy_python/leet_088/leet_088a.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        pos = m+n-1
        m = m-1
        n = n-1
        while pos >=0:
            if n<0: return
            if m<0 or nums1[m] < nums2[n]: 
                nums1[pos] = nums2[n]
                n-=1
            else:
                nums1[pos] = nums1[m]
                m-=1
            pos-=1    
            
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: void Do not return anything, modify nums1 in-place instead.
        """
        